### DISCUSSION TITLE
Update minimal python version up to 3.10

### DISCUSSION THREAD

**Discussion by d-a-bunin** [etna-team/etna#420] [Improvements]
### Description

Currently (2.7.1) the library supports python versions: (3.8, 3.9, 3.10).
- We can't update to higher versions due to dependency conflict (`pytorch_forecasting`)
- We use `typing` module extensively, but on new versions of python we could use expressions like `list[int]` instead of `typing.List[int]`
- Old python versions could prohibit us from adding/updating some of the dependencies, because e.g. they could require python > 3.8.

### Thoughts

I'm not sure if it is necessary to support the old python versions like 3.8 and 3.9 after the major update.

### Related

- TODO: link discussion about `pytorch_forecasting` after creation

**Comment by xeonvs** [comment]
Latest versions(1.0+) of pytorch_forecasting now support python 3.1[12] and pandas <3.0

**Reply by d-a-bunin** [reply]
Currently we have a dependency `pytorch_forecasting=^0.9.0"` (which means >= 0.9.0 and <1). After updating to v1 some breaking changes in a dependencies were made (e.g. optuna changed from v2 to v3), so it is difficult for us to support both <1 and >=1 for `pytorch_forecasting`

**Comment by egoriyaa** [comment]
Solved in #599
